esp/platform/txsummary.hpp

```diff
@@ -50,80 +75,264 @@ class CTxSummary : extends CInterface
     virtual unsigned getElapsedTime() const;
 
     // Appends all summary entries to the given buffer.
-    virtual void serialize(StringBuffer& buffer, const LogLevel logLevel = LogMin) const;
-
+    //
+    // JSON serialization supports string, integer numeric, true, false,
+    // null, and object values. Arrays are not supported.
+    virtual void serialize(StringBuffer& buffer, ITxSummaryProfile* profile, const LogLevel logLevel = LogMin, const unsigned int group = DEFAULT_GROUP, const unsigned int requestedStyle = TXSUMMARY_OUT_TEXT) const;
     // Adds the unique key and value to the end of the summary.
     // Returns true if the key value pair are added to the summary. Returns
     // false if the key is NULL, empty, or not unique within the summary.
-    virtual bool append(const char* key, const char* value, const LogLevel logLevel = LogMin);
-    template <typename TValue, typename TSuffix = const char*, class TSerializer = TokenSerializer>
-    bool append(const char* key, const TValue& value, const LogLevel logLevel = LogMin, const TSuffix& suffix = "", const TSerializer& serializer = TSerializer());
```

Comment:
Removing this form of append does not affect us internally. But it has existed since 6.0, which means there is a risk (however slight it may be) this could be a breaking change for somebody. The templated serializer permits any construct for which a string representation exists to be appended without first assembling the string.

Might not be an issue for anybody. Workarounds exist if it is. I'm just pbserving the potential impact.

Reply:
reinstated template forms